glusterfsd/src/glusterfsd-mgmt.c

```diff
@@ -2604,6 +2604,8 @@ glusterfs_volfile_fetch_one(glusterfs_ctx_t *ctx, char *volfile_id)
                               GF_HNDSK_GETSPEC, mgmt_getspec_cbk,
                               (xdrproc_t)xdr_gf_getspec_req);
 
+    /*  In case of error the frame will be destroy by rpc_clnt_submit */
+    frame = NULL;
```

Comment:
There are some cases where `mgmt_submit_request()` may return an error but the callback is not executed. In those cases the frame would leak with this change.

Reply:
Yes it is true but i think we can ignore those (iobref_new, iobuf_get2, xdr_serialize_generic) because if these functions will fail it would be difficult to survive a process.

Reply:
Errors in some functions may make difficult to do useful work, but unhandled errors will surely end in a crash sooner or later, which is the worst case. IMO either we abort when something that we believe can never happen, or that we can't recover from it, actually happens, or we need to do correct error handling. Note that recovering from an error just means that Gluster is still operational and in a consistent state, even if the errors don't allow to do real work (i.e. the application will see those errors, but it will still be able to continue accessing the filesystem)

Reply:
ok, I will send a new patch.